crates/turborepo-scm/src/manual.rs

```diff
@@ -109,7 +111,7 @@ pub(crate) fn get_package_file_hashes_from_processing_gitignore<S: AsRef<str>>(
         if metadata.is_symlink() {
             continue;
         }
```

Comment:
This line of code + `follow_links = false` means that if the symlink target is outside of the package or in the package but otherwise ignored we get a bad hash.

I believe that we should be following symlinks, and if it is a broken symlink, hash it the same way `git` does (by target).

Reply:
(This would also means that if we `ls-tree` and have a symlink, we need to replace it with the target hash if the target exists, and, if the target is a directory, walk it.)

Reply:
The intention with the change was to hash symlink files, not their targets. There's been a series of differently-incorrect behavior here:
 - As @arlyon notes, `ls-tree` does list a hash for the symlink. So without inputs, they are wholesale ignored.
 - _when using inputs_, and at least since switching to the encoder setup in Go, symlinks caused an error and we fell back to manual hashing. It's very possible that was the behavior before also, I haven't checked
 - manual hashing in Go calls `os.Open`, which reads the target.
 - Rust `ls-tree` has the same restriction as the Go implementation, it ignores symlinks
 - Rust git-based hashing, _with inputs_, was intended to switch to hashing the contents of the symlink, but there are two problems:
   1. We eventually get to `fopen` as @nathanhammond notes. This follows the link. The Go manual file hashing implementation has the same behavior. At least for the Rust implementation, that was not the intention, and I would consider it a bug at least insofar as it doesn't do what was intended.
   2. As noted in the slack thread, the _raw_ contents of the link is not portable, so probably not what we want to hash

In terms of what I think _should_ happen is the following:
 - Hash a portable representation of the contents of the symlink, _not_ the target. This saves us from having to care about file vs directory, broken vs working, and in-repo vs not, which in the presence of `inputs` is not even necessarily a clear distinction (do we follow in-repo, but out-of-inputs links?). Users will be required to ensure that the target is covered by their inputs if they want the target hashed. This is mirrors our behavior for caching: we don't cache the _target_ of symlinks. If you want it cached, ensure that the target is also covered by the `outputs` globs. 
 - Not for this PR, but eventually: determine how to handle symlinks when `inputs` is not specified.